### PR TITLE
Update upgrade command in preparation for release

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs

This PR updates the `nebari upgrade` command to improve the UI and to fix the "image update" functions. Lastly, I updated the versions for `nebari-dask`, all the docker images and the `nebari-workflow-controller`.

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: src/_nebari/upgrade.py [hunk 1]
```diff
@@ -141,12 +142,10 @@ def upgrade_step(self, config, start_version, config_filename, *args, **kwargs):
         for any actions that are only required for the particular upgrade you are creating.
         """
         finish_version = self.get_version()
-        __rounded_finish_version__ = ".".join(
-            [str(c) for c in rounded_ver_parse(finish_version)]
-        )
+        ".".join([str(c) for c in rounded_ver_parse(finish_version)])
```
Comment (costrouc) [inline src/_nebari/upgrade.py:145]: ` __rounded_finish_version__` was this variable dropped?

File: src/_nebari/upgrade.py [hunk 2]
```diff
@@ -408,18 +411,41 @@ def _version_specific_upgrade(
...
+            response = Prompt.ask(
+                "\nDo you want to enable the [green][link=https://nebari-docs.netlify.app/docs/how-tos/using-argo#jupyterflow-override-beta]Nebari Workflow Controller[/link][/green], required for [green][link=https://github.com/nebari-dev/argo-jupyter-scheduler]Argo-Jupyter-Scheduler[/link][green]? [Y/n] ",
+                default="Y",
+            )
+            if response.lower() in ["y", "yes", ""]:
+                argo["nebari_workflow_controller"] = {"enabled": True}
+
+        rich.print("\n ⚠️ Deprecation Warnings ⚠️")
+        rich.print(
+            f"-> [green]{self.version}[/green] is the last Nebari version that supports CDS Dashboards"
```
Comment (costrouc) [inline src/_nebari/upgrade.py:443]: Thanks for adding this to the notes.

**Review by costrouc — APPROVED** [review]
LGTM. And thanks for making the messaging better than it was before.

File: src/_nebari/upgrade.py [hunk 3]
```diff
@@ -408,18 +411,41 @@ def _version_specific_upgrade(
...
+class Upgrade_2023_7_1(UpgradeStep):
+    version = "2023.7.1"
+
+    def _version_specific_upgrade(
+        self, config, start_version, config_filename: Path, *args, **kwargs
+    ):
+        argo = config.get("argo_workflows", {})
+        if argo.get("enabled"):
+            response = Prompt.ask(
+                "\nDo you want to enable the [green][link=https://nebari-docs.netlify.app/docs/how-tos/using-argo#jupyterflow-override-beta]Nebari Workflow Controller[/link][/green], required for [green][link=https://github.com/nebari-dev/argo-jupyter-scheduler]Argo-Jupyter-Scheduler[/link][green]? [Y/n] ",
```
Comment (aktech) [inline src/_nebari/upgrade.py:435]: I would suggest make the links as variables and using them here.

File: src/_nebari/upgrade.py (hunk 1 above)
Comment (iameskild) [reply src/_nebari/upgrade.py]: Yeah I should be using this... I added it back.

**Pull request merged**